### PR TITLE
Sqlalchemy experiments

### PR DESCRIPTION
**Pull request by Valeria1235**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Support ydb dialect for sqlalchemy. Work in progress.

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

Issue Number: https://github.com/ydb-platform/ydb-python-sdk/issues/111

### CODE REVIEW

**Review by rekby — CHANGES_REQUESTED** [review]
and need to pass tests

File: ydb/_dbapi/__init__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    DatabaseError,
+    DataError,
+    OperationalError,
+    IntegrityError,
+    InternalError,
+    ProgrammingError,
+    NotSupportedError,
+)
+
+version = "0.0.31"
```
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_dbapi/__init__.py:15]: I have no idea) will drop it

File: ydb/_sqlalchemy/__init__.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    def dbapi():
+        return dbapi
+
+    def _check_unicode_returns(self, *args, **kwargs):
+        # Normally, this would do 2 SQL queries, which isn't quite necessary.
+        return "conditional"
+
+    def get_columns(self, connection, table_name, schema=None, **kw):
+        if schema is not None:
+            raise dbapi.errors.NotSupportedError
```
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_sqlalchemy/__init__.py:259]: this isn't necessary

File: ydb/_dbapi/connection.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    def check_exists(self, table_path):
+        try:
+            self.driver.scheme_client.describe_path(table_path)
+            return True
+        except ydb.SchemeError:
+            return False
+
+    def commit(self):
+        pass
```
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_dbapi/connection.py:39]: no, sqlalchemy used it implicitly and there will raises on `engine.connect`

File: ydb/_dbapi/connection.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            driver.wait(timeout=5, fail_fast=True)
+        except ydb.Error as e:
+            raise DatabaseError(e.message, e.issues, e.status)
+        except Exception:
+            driver.stop()
+            raise DatabaseError(
+                f"Failed to connect to YDB, details {driver.discovery_debug_details()}"
+            )
+
+        return driver, ydb.SessionPool(driver)
```
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_dbapi/connection.py:75]: i agree

**Review by rekby — APPROVED** [review]

**Pull request merged**